src/query/functions/src/srfs/variant.rs

```diff
@@ -460,6 +470,162 @@ pub fn register(registry: &mut FunctionRegistry) {
+
+                            match json_arg.index(row) {
+                                Some(ScalarRef::Null) => {
+                                    return null_result;
+                                }
+                                Some(ScalarRef::Variant(v)) => {
+                                    let s = to_serde_json(v);
+                                    match s {
+                                        Err(e) => {
+                                            ctx.set_error(row, e.to_string());
+                                            return null_result;
+                                        }
+                                        Ok(s) => {
+                                            let jaq_val = Val::from(s);
+                                            let jaq_out = filter.run((jaq_ctx.clone(), jaq_val));
+
+                                            for res in jaq_out {
+                                                match res {
+                                                    Err(err) => {
+                                                        ctx.set_error(row, err.to_string());
+                                                        return null_result;
+                                                    }
+                                                    Ok(res) => {
+                                                        let res_json_str = jaq_val_to_json(&res);
+                                                        res_builder.put_str(&res_json_str);
+                                                        res_builder.commit_row();
+                                                    }
+                                                };
+                                            }
+                                        }
+                                    }
+                                }
+                                None => {
+                                    return null_result;
+                                }
+                                _ => unreachable!(),
+                            }
+
+                            let res_col = Column::Variant(res_builder.build()).wrap_nullable(None);
+                            let res_len = res_col.len();
+                            max_nums_per_row[row] = std::cmp::max(max_nums_per_row[row], res_len);
+                            (Value::Column(Column::Tuple(vec![res_col])), res_len)
+                        })
+                        .collect()
+                }),
+            },
+        }))
+    });
+}
+
+// This comes straight from the jaq source. It converts a Jaq val type to a JSON string.
+// https://github.com/01mf02/jaq/blob/426fdab46c95e7ed0dadc5c049b3d83388271b1a/jaq/src/main.rs#L511
+// there may be an opportunity to format directly into
+// jsonb data so it doesn't need to be re-parsed.
+fn jaq_val_to_json(val: &Val) -> String {
+    match val {
+        Val::Null => "null".to_string(),
```

Comment:
We can convert `Val` to `jsonb::Value` and then convert to binary format.
for `Array` and `Object`, we can use function `jsonb::build_array` and `jsonb::build_object`.
[cast_scalar_to_variant](https://github.com/datafuselabs/databend/blob/main/src/query/expression/src/types/variant.rs#L198) function is an example to look at.